AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -183,6 +183,81 @@ + (void)deleteDocumentWithPartition:(NSString *)partition
 }
 
 #pragma mark - MSDataStore Implementation
+
+- (void)readWithPartition:(NSString *)partition
+               documentId:(NSString *)documentId
+             documentType:(Class)documentType
+              readOptions:(MSReadOptions *)__unused readOptions
+        completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler {
+
+  // TODO consume writeOptions
+  [MSTokenExchange
+      performDbTokenAsyncOperationWithHttpClient:(MSStorageIngestion *)self.ingestion
+                                       partition:partition
+                               completionHandler:^(MSTokensResponse *_Nonnull tokenResponses, NSError *_Nonnull error) {
+                                 // If error getting token.
+                                 if (error || !tokenResponses) {
+                                   MSLogError([MSDataStore logTag], @"Can't get CosmosDb token:%@", [error description]);
+                                   completionHandler([[MSDocumentWrapper alloc] initWithError:error documentId:documentId]);
+                                   return;
+                                 }
+
+                                 // Create http client.
+                                 MSCosmosDbIngestion *cosmosDbIngestion = [[MSCosmosDbIngestion alloc] init];
```

Comment:
It looks like there is already an ingestion property for this class. Why not use it? (same question about the code this is copied from)

Reply:
this will change into a single http client after the http refactor work, currently we need 2 separate http clients(ingestion) for token and cosmos(since requests are different)

Reply:
But there is a property that isn't being used, and I fear that if we don't use it, it will be left as dead code. It shouldn't  be more than a couple of lines to change this now, right?